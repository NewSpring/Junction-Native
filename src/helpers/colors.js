// @flow
import Style from "further";

import createHelpers from "./util/createHelpers";
import colors from "./util/modifiers/colors";
import combine from "./util/combine";

export default createHelpers(colors, { property: "color" }).reduce(
  combine,
  Style.empty(),
);
