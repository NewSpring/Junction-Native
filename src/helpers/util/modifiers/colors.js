//@flow

import type IHelperProperty from "../createHelpers";

export const reducer = (properties: IHelperProperty[]) =>
  properties.map(({ name, property, props }) => ({
    [property]: props[property] ? props[property] : props.theme.colors[name],
  }));

export default [
  { name: "primary", reducer },
  { name: "secondary", reducer },
  { name: "tertiary", reducer },
  { name: "dark", reducer },
  { name: "darkSecondary", reducer },
  { name: "darkTertiary", reducer },
  { name: "light", reducer },
  { name: "lightSecondary", reducer },
  { name: "lightTertiary", reducer },
];
