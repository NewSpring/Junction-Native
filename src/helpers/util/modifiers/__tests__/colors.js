import colors, { reducer } from "../colors";

it("has a reducer which pulls from props", () => {
  const result = reducer([
    { name: "blue", property: "test", props: { test: "1" } },
  ])[0];
  expect(result).toEqual({ test: "1" });
});

it("has a reducer which pulls from the theme", () => {
  const result = reducer([
    {
      name: "blue",
      property: "test",
      props: { theme: { colors: { blue: "1" } } },
    },
  ])[0];
  expect(result).toEqual({ test: "1" });
});

it("has 9 colors", () => {
  expect(colors).toMatchSnapshot();
});
