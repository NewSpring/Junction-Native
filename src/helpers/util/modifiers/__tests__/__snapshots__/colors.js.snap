// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has 9 colors 1`] = `
Array [
  Object {
    "name": "primary",
    "reducer": [Function],
  },
  Object {
    "name": "secondary",
    "reducer": [Function],
  },
  Object {
    "name": "tertiary",
    "reducer": [Function],
  },
  Object {
    "name": "dark",
    "reducer": [Function],
  },
  Object {
    "name": "darkSecondary",
    "reducer": [Function],
  },
  Object {
    "name": "darkTertiary",
    "reducer": [Function],
  },
  Object {
    "name": "light",
    "reducer": [Function],
  },
  Object {
    "name": "lightSecondary",
    "reducer": [Function],
  },
  Object {
    "name": "lightTertiary",
    "reducer": [Function],
  },
]
`;
